crates/ide_assists/src/handlers/replace_derive_with_manual_impl.rs

```diff
@@ -52,18 +53,34 @@ pub(crate) fn replace_derive_with_manual_impl(
         return None;
     }
 
-    let trait_token = args.syntax().token_at_offset(ctx.offset()).find(|t| t.kind() == IDENT)?;
-    let trait_name = trait_token.text();
-
+    let current_derives = args.clone().as_comma_sep_paths()?;
+    // found the related path of current focused token
+    // I don't know how to keep the original range for tokens while transforming and parsing as new ast
+    let mut offset =
+        args.syntax().descendants_with_tokens().find(|t| t.kind() == IDENT)?.text_range().start();
+    let trait_path = current_derives.iter().enumerate().find_map(|(index, path)| {
+        let comma_offset = if index == 0 { "" } else { "," };
+        let current_range = path
+            .syntax()
+            .text_range()
+            .checked_add(offset)?
+            .checked_add(TextSize::of(comma_offset))?;
+        offset = current_range.end();
+        let in_range = current_range.start() <= ctx.offset() && ctx.offset() <= current_range.end();
+        if in_range {
+            Some(path)
+        } else {
+            None
+        }
+    })?;
     let adt = attr.syntax().parent().and_then(ast::Adt::cast)?;
 
     let current_module = ctx.sema.scope(adt.syntax()).module()?;
     let current_crate = current_module.krate();
-
     let found_traits = items_locator::items_with_name(
         &ctx.sema,
         current_crate,
-        NameToImport::Exact(trait_name.to_string()),
+        NameToImport::Exact(trait_path.segments().last()?.to_string()),
```

Comment:
An aside(not relevant for current PR): it would be interesting to have this assist search through the derive attribute's expansion for trait impls to figure out which trait import to show.